src/core/positioning/egenioussreceiver.cpp

```diff
@@ -36,17 +38,74 @@ EgenioussReceiver::EgenioussReceiver( QObject *parent )
 EgenioussReceiver::~EgenioussReceiver()
 {
   mTcpSocket->deleteLater();
+  mNetworkManager->deleteLater();
   mTcpSocket = nullptr;
+  mNetworkManager = nullptr;
 }
 
 void EgenioussReceiver::handleConnectDevice()
 {
-  mTcpSocket->connectToHost( mAddress, mPort, QTcpSocket::ReadWrite );
+  QNetworkRequest request( QString( "http://%1:%2/app/start" ).arg( mAddress.toString() ).arg( mPort ) );
+  mNetworkManager->get( request );
 }
 
 void EgenioussReceiver::handleDisconnectDevice()
 {
-  mTcpSocket->disconnectFromHost();
+  QNetworkRequest request( QString( "http://%1:%2/app/stop" ).arg( mAddress.toString() ).arg( mPort ) );
+  mNetworkManager->get( request );
+}
+
+void EgenioussReceiver::onHttpFinished( QNetworkReply *reply )
```

Comment:
OK, here let's simplify things here and let's use the QNetworkReply::finished signal instead using in-place functions within the handleConnectDevice and handleDisconnectDevice. That'll avoid a massive function that handles all requests.